core/vm/evm.go

```diff
@@ -221,7 +234,7 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 		return nil, gas, ErrDepth
 	}
 	// Fail if we're trying to transfer more than the available balance
-	if !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
+	if value.Sign() != 0 && !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
```

Comment:
is `value.Sign() != 0` necessary? I don't see this in ethereum https://github.com/ethereum/go-ethereum/pull/21672

Reply:
It's in #21222:

[```](https://github.com/holiman/go-ethereum/blob/8528951f5477600ce703bdc37b1e3071b5619b8c/core/vm/evm.go#L204-L206)